File: codinGame/FaxMachine/FaxMachine.js
const w = parseInt(readline());
const h = parseInt(readline());
const t = readline();

let s = t.split(' ')

let r = []
for (let i=0; i<s.length; i++)
{
    if (i%2 == 0)
        // for x in range(int(r[i])):
        for (let x=0; x<parseInt(s[i]); x++)
            r.push('*')
    else
        for (let x=0; x<parseInt(s[i]); x++)
            r.push(' ')
}

let x=0
let i=0

while (i<r.length)
{
    if (!x)
        process.stdout.write('|')
    if (x<w)
    {
        process.stdout.write(r[i])
        x++
        i++
    }
    else
    {
        console.log('|')
        x=0
    }
}
console.log('|')

File: codinGame/FaxMachine/FaxMachine.py
w = int(input())
h = int(input())
t = input()
r = t.split(' ')

t = []
for i in range(len(r)):
    if not i%2:
        for x in range(int(r[i])):
            t.append('*')
    else:
        for x in range(int(r[i])):
            t.append(' ')

x=0
i=0

while i<len(t):
    if not x:
        print('|', end='')
    if x<w:
        print(t[i], end='')
        x+=1
        i+=1
    else:
        print('|')
        x=0
print('|')

